### PR TITLE
BCN - Mark and Jordi

### PR DESCRIPTION
**Pull request by markvertra**
There is a lot to refactor, and we have taken advantage of the fact that the list of songs in Abbey Road already came ordered BUT we used Lodash methods like _.last. So that's that.

### CODE REVIEW

**Comment by markvertra** [comment]
@jviles

File: starter-code/lib/exercises.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+var employeeSatisfaction = function() {
+	
+	function randomOpinion(array) {
+	return _.sample(array);
+	}
+
+	function arrayOpinions(num){
+	var resultArray = [];
+	for (i = 0; i < num; i++) {
```
Comment (tawebbcn) [inline starter-code/lib/exercises.js:15]: Better use _.times()

File: starter-code/lib/exercises.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		resultArray.push(randomOpinion(opinions));
+	}
+	return resultArray;
+}
+
+	function matrixOpinions(num){
+	var resultArray2 = [];
+	var resultArray1 = _.concat(resultArray2, [arrayOpinions(10)], 
+		[arrayOpinions(10)], [arrayOpinions(10)], [arrayOpinions(10)], [arrayOpinions(10)])
```
Comment (tawebbcn) [inline starter-code/lib/exercises.js:24]: You do the same 5 times... I wonder if there is a Lodash function for that...

File: starter-code/lib/exercises.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ }
+};
+
+// noRepeatChar(goodPsswd);
+// noRepeatChar(badPsswd);
+
+var goodPsswd = "1234567890";
+var badPsswd = "1a234567890";
+var onlyNumbers = function (password) {
+	var numbers = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"]
```
Comment (tawebbcn) [inline starter-code/lib/exercises.js:51]: You have methods in Lodash as _.isNan() and _.toNumber.

File: starter-code/lib/exercises.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	} else {
+		console.log("Good password")
+	}
+}
+// onlyNumbers(goodPsswd);
+// onlyNumbers(badPsswd);
+
+var goodPsswd = "1234567890";
+var badPsswd = "12345678901234567890";
+var trimPassword = function (password) {
```
Comment (tawebbcn) [inline starter-code/lib/exercises.js:65]: You also have _.dropRight(), but this is ok.

File: starter-code/lib/harryPotter.js [hunk 5]
```diff
@@ -8,3 +8,34 @@
...
+// 		}
+// 		return arrayEven, arrayOdd;
+// 	})
+// 	return array1, array2;
+// }
+
+
+function potterSplitter(info) {
+	var array = _.chunk(info, 2);
+	return _.zip(array)
```
Comment (tawebbcn) [inline starter-code/lib/harryPotter.js:32]: You don't need the _.zip()

File: starter-code/lib/abbeyRoad.js [hunk 6]
```diff
@@ -911,3 +911,55 @@ var abbeyRoadRecords = [{song: "Land of Hope and Glory",
                       {song: "Love Divine III",
                       artist: "Jan Mulder with The Royal Philharmonic Orchestra",
                       year: 2016}];
+
+var novemberArtists = function (array) {
+  var discos = array.filter(function(object){
```
Comment (tawebbcn) [inline starter-code/lib/abbeyRoad.js:916]: You also have a _.filter() method.

**Comment by tawebbcn** [comment]
About the order, you have methods like _.orderBy().

Better put all exercises in exercises.js and don't add anything to the data files.

That said, good job, congratulations! :+1:

**Pull request closed without merging**